persistence/sql/src/main/java/org/infinispan/persistence/sql/AbstractSchemaJdbcStore.java

```diff
@@ -0,0 +1,626 @@
+      if (parameterMap.size() - (hasEmbeddedKey ? 0 : uniquePrimaryParameters) > 1 || valueMessageName != null) {
+         if (valueMessageName == null || packageName == null) {
+            throw new CacheConfigurationException("Table has multiple non key columns but no value message schema defined, which is required when there is more than one value column");
+         }
+         String fullMessageName = packageName + "." + valueMessageName;
+         verifyParametersPresentForMessage(fullMessageName, parameterMap, false);
+         this.valueMessageName = fullMessageName;
+      } else {
+         // This variable is only for assertion - it should be that we can only have 1 non primary parameter,
+         // but just in case
+         boolean updatedPrimitive = false;
+         for (Parameter parameter : parameters) {
+            if (parameter.primaryIdentifier) {
+               continue;
+            }
+            assert !updatedPrimitive;
+            updatePrimitiveJsonConsumer(parameter, false);
+            updatedPrimitive = true;
+         }
+      }
+
+      List<Parameter> unusedValueParams = null;
+
+      for (Parameter parameter : parameters) {
+         if (parameter.jsonUpdateConsumer == null) {
+            if (parameter.primaryIdentifier) {
+               // This is a misconfigured schema to table
+               throw new CacheConfigurationException("Primary key " + parameter.name + " was not found in the schema!");
+            } else {
+               if (unusedValueParams == null) {
+                  unusedValueParams = new ArrayList<>();
+               }
+               unusedValueParams.add(parameter);
+            }
+         }
+      }
+
+      if (unusedValueParams != null) {
+         parameters = handleUnusedValueParams(parameters, unusedValueParams);
+      }
+
+      if (hasEmbeddedKey) {
+         // Make sure all values are mapped as they must be when embedded key
+         assert Arrays.stream(parameters).noneMatch(parameter -> parameter.unwrapJsonValue == null);
+      } else {
+         // Primary identifiers shouldn't have any values mapped as they aren't embedded
+         assert Arrays.stream(parameters).noneMatch(parameter -> parameter.primaryIdentifier && parameter.unwrapJsonValue != null);
+         assert Arrays.stream(parameters).noneMatch(parameter -> !parameter.primaryIdentifier && parameter.unwrapJsonValue == null);
+      }
+
+      assert Arrays.stream(parameters).filter(Parameter::isPrimaryIdentifier).noneMatch(parameter -> parameter.unwrapJsonKey == null);
+   }
+
+   Parameter[] handleUnusedValueParams(Parameter[] parameters, List<Parameter> unusedValueParams) {
+      // If it is a loader, we can ignore missing values as it is read only
+      if (!config.ignoreModifications()) {
+         throw unusedValueParamsException(unusedValueParams);
+      }
+      // TODO: add debug message here to say that columns are missing from schema object
```

Comment:
Shouldn't we throw an exception here? If columns are missing from the schema this is probably a mistake from the user .. or is this intentional so that users can avoid specifying all fields in tables with many columns?